### PR TITLE
FIx for #356 - support for configuring routes in pom

### PR DESCRIPTION
**Pull request by sejacobsen**
## Description

Add further support for tls termination and insecureEdgeTerminationPolicy in pom.xml

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
~- [ ] I tested my code in Kubernetes~
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java [hunk 1]
```diff
@@ -173,4 +172,41 @@ public void testCreateWithDefaultsAndExistingRouteWithMatchingNameInBuilderInOpe
...
+    }
+
+    @Test
+    public void testEnrichWithTls(){
+        // Given
+        properties.put("jkube.openshift.generateRoute.tls.termination", "edge");
+        properties.put("jkube.openshift.generateRoute.tls.insecure_edge_termination_policy", "Allow");
+        new RouteEnricher(context).create(PlatformMode.openshift, klb);
+        // When
+        new RouteEnricher(context).enrich(PlatformMode.openshift, klb);
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java:201]: Umm, I think it's just my personal opinion. But would it be nice to do it using only single instance of RouteEnricher?

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java [hunk 2]
```diff
@@ -50,11 +49,16 @@
...
     private static final String GENERATE_ROUTE_PROPERTY = "jkube.openshift.generateRoute";
+    private static final String GENERATE_TLS_TERMINATION_PROPERTY = "jkube.openshift.generateRoute.tls.termination";
+    private static final String GENERATE_TLS_INSECURE_EDGE_TERMINATION_POLICY_PROPERTY = "jkube.openshift.generateRoute.tls.insecure_edge_termination_policy";
     public static final String EXPOSE_LABEL = "expose";
 
     @AllArgsConstructor
     private enum Config implements Configs.Config {
-        GENERATE_ROUTE("generateRoute", "true");
+        GENERATE_ROUTE("generateRoute", "true"),
+        //TARGET_PORT("targetPort", "8080"),
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:59]: You might have forgotten to remove this.

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java [hunk 3]
```diff
@@ -50,11 +49,16 @@
 public class RouteEnricher extends BaseEnricher {
 
     private static final String GENERATE_ROUTE_PROPERTY = "jkube.openshift.generateRoute";
+    private static final String GENERATE_TLS_TERMINATION_PROPERTY = "jkube.openshift.generateRoute.tls.termination";
+    private static final String GENERATE_TLS_INSECURE_EDGE_TERMINATION_POLICY_PROPERTY = "jkube.openshift.generateRoute.tls.insecure_edge_termination_policy";
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:53]: Unless there is a specific reason to include custom properties (e.g. compatibility with other systems, libraries, etc.) the recommended approach is to use the built-in property & configuration system.

In this case, both your configurations are configurable by the use of the following properties:
```
jkube.enricher.jkube-openshift-route.termination
jkube.enricher.jkube-openshift-route.insecureEdgeTerminationPolicy
```

Please, if there is no reason to add this extra complexity, remove these properties and make use of the standard approach.

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java [hunk 4]
```diff
@@ -72,6 +76,11 @@ private boolean isGenerateRoute() {
         return Configs.asBoolean(getConfigWithFallback(Config.GENERATE_ROUTE, GENERATE_ROUTE_PROPERTY, null));
     }
 
+    private boolean isRouteWithTLS(){
+        String propval = getConfigWithFallback(Config.TLS_TERMINATION, GENERATE_TLS_TERMINATION_PROPERTY, null);
+        return (propval != null && !propval.isEmpty());
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:81]: Please Use `StringUtils#isNotBlank`

**Comment by manusa** [comment]
You'll need to rebase this branch onto master before we can merge it.

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java (hunk 4 above)
Comment (sejacobsen) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:80]: Sorry, didn't see these comments. Will fix and rebase

**Comment by sejacobsen** [comment]
Sorry, did not see latest comments - will fix and rebase

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java (hunk 3 above)
Comment (sejacobsen) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:53]: Of course - was not aware that these existed. Can you point me to an overview of existing properties? Can't seem to find these in the project

Comment (sejacobsen) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:53]: @manusa  now I think I get your point - follow the pom layout for property naming? 
In that case, I suppose also the GENERATE_ROUTE_PROPERTY should be jkube.enricher.jkube-openshift-route.generateRoute?

Comment (manusa) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java:53]: Yep, you can check that in the docs (Line 228):
https://github.com/eclipse/jkube/blob/7e42f3bad9bec43d82a2425e89d9b1dc94408c8a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc#L223-L228

You **don't** need to provide any property resolution yourself, `getConfig` method takes care of that for you.

The other property has been kept for backwards compatibility with FMP, but this is not what we want for new properties.

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java (hunk 1 above)
Comment (sejacobsen) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java]: @rohanKanojia  I agree. I have removed the enrich method in RouteEnricher entirely, so I can keep the structure of tests and method invocations as-is.

**Comment by sejacobsen** [comment]
@rohanKanojia don't know about the failing test? Can I retest without pushing changes?

**Comment by rohanKanojia** [comment]
I've restarted the workflow

**Comment by rohanKanojia** [comment]
@sejacobsen: Could you please rebase your PR with the latest master? I see some other commits in your PR too. Could you please squash your commits after rebasing?

**Comment by sejacobsen** [comment]
I think I have messed around a little too much on this one. Will close PR and open another. Sorry about the trouble, will include suggestions above to the best of my ability :)

**Pull request closed without merging**